packages/core-design-system/scripts/build.js

```diff
@@ -0,0 +1,304 @@
+import fs from 'node:fs/promises'
+
+import { permutateThemes, register } from '@tokens-studio/sd-transforms'
+import StyleDictionary from 'style-dictionary'
+import { getReferences, usesReferences } from 'style-dictionary/utils'
+
+import { harnessLog } from './complete-log.js'
+import {
+  COMMON_CONFIG,
+  DESIGN_SYSTEM_PREFIX,
+  DESIGN_SYSTEM_ROOT,
+  DESIGN_SYSTEM_ROOT_ESM,
+  getExportFileHeader,
+  OSS_STYLES_SOURCE_NAME,
+  STYLE_BUILD_FORMATS,
+  THEME_MODE_FILENAME_PREFIX
+} from './constants.js'
+import { generateCoreFiles, generateThemeFiles } from './sd-file-generators.js'
+
+register(StyleDictionary)
+
+async function run() {
+  let $themes
+  try {
+    $themes = JSON.parse(await fs.readFile('design-tokens/$themes.json'))
```

Comment:
Not a big deal, but node gets annoyed when you don't specify the file encoding and returns a buffer instead of a string. `JSON.parse` handles that, but it's not a bad idea to be specific.
```suggestion
  $themes = JSON.parse(await fs.readFile('design-tokens/$themes.json', { encoding: 'utf8' }))
```

Or just

```suggestion
  $themes = JSON.parse(await fs.readFile('design-tokens/$themes.json', 'utf8'))
```

I _personally_ prefer the former to be super-clear, but either works.